File: content-solutions/sorting/score-ranks.py
def solution(scores):
    count = [0] * 101
    for s in scores:
        count[s] += 1
    higher = [0] * 101
    for v in range(99, -1, -1):
        higher[v] = higher[v + 1] + count[v + 1]
    return [higher[s] + 1 for s in scores]

File: content-solutions/sorting/score-ranks.js
function solution(scores) {
  const count = Array(101).fill(0);
  for (const s of scores) count[s]++;
  const higher = Array(101).fill(0);
  for (let v = 99; v >= 0; v--) higher[v] = higher[v + 1] + count[v + 1];
  return scores.map((s) => higher[s] + 1);
}
